Enhancement #169 - Add UI filtering of discussions and markers

omerxx opened the pull request:
#169 - Added UI filtering to show only discussions or only markers.
This doesn't refer to the sidebar which shows only accidents anyhow.
On `clusterMode()` the field is disabled, and enabled again when zoomed in.

UI created with the kind help of @LironRS

njenia commented:
2 things:
- It seems that in markers() you call bounding_box_query() twice: once for each marker type (regular and discussion). This method is already a performance bottleneck and I would prefer not to overload it. You should place the check that you have in bounding_box_query() outside so that it would be called only once.
- It's not best practice to put <script> tags with actual code inside html files. Presentation and logic should be separated. Place it in an appropriate js file.

omerxx commented:
Talked to @njenia , fix is ready to be merged

danielhers commented:
Please rebase to `dev` so we can merge this. If this was already merged to `master`, please close because we merged `master` to `dev` already.

omerxx commented:
Ready to merge.

@danielhers

danielhers commented on main.py:
```diff
@@ -110,15 +110,17 @@ def markers():
     severe = int(request.values['show_severe'])
     light = int(request.values['show_light'])
     inaccurate = int(request.values['show_inaccurate'])
+    mark_on = bool(request.values['mark_on'])
```
Generally it's better to use complete words, since clarity is more important than saving bits. Also, why `_on` and not the former convention of `show_`?

danielhers commented on static/js/app.js:
```diff
@@ -754,4 +758,31 @@ $(function () {
+    switch (li) {
+        case 1:
+            mark_on='1';disc_on='1';
+            document.getElementById("view-filter").value = "הצג הכל";
```
Why not jQuery here?

danielhers commented on static/js/app.js:
```diff
@@ -754,4 +758,31 @@ $(function () {
+
+function load(li) {
+    switch (li) {
+        case 1:
```
Please use constants for there magic numbers

danielhers commented on templates/index.html:
```diff
@@ -335,13 +335,35 @@ <h4 class="title"></h4>
+                    <div class="input-append dropdown">
+                        <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+                            <span class="add-on" style="border: 1px solid #ccc; border-radius: 3px 3px 0 0; margin-left:-1px"><i class="icon-filter"></i></span>
+                            <input id="view-filter" value="הצג את כלל התאונות והדיונים" style="padding-right:7px; border: 1px solid #ccc; height:26px; width:204px; border-radius: 0 0 3px 3px; position:absolute;"/>
```
כלל should be כל?

galraij replied:
You can just write: הצג תאונות ודיונים



‫ב-1 ביולי 2015, בשעה 06:33, ‏‏Daniel Hershcovich ‏<notifications@github.com> כתב/ה:‬

> In templates/index.html:
> 
> > @@ -335,13 +335,35 @@ <h4 class="title"></h4>
> >                  </li>
> >  
> >              </ul>
> > +
> > +            <div class="sidebar-title">סינון דיונים ותאונות</div>
> > +            <ul>
> > +                <li style="height: 30px">
> > +                    <div class="input-append dropdown">
> > +                        <a class="dropdown-toggle" data-toggle="dropdown" href="#">
> > +                            <span class="add-on" style="border: 1px solid #ccc; border-radius: 3px 3px 0 0; margin-left:-1px"><i class="icon-filter"></i></span>
> > +                            <input id="view-filter" value="הצג את כלל התאונות והדיונים" style="padding-right:7px; border: 1px solid #ccc; height:26px; width:204px; border-radius: 0 0 3px 3px; position:absolute;"/>
> כלל should be כל?
> 
> —
> Reply to this email directly or view it on GitHub.
>

omerxx commented:
Changed 
`mark_on` to `show_markers`
`disc_on` to `show_discussions`

Updated
`document.getElementById` to jQuery
switch cases variables names in `app.js` 

Corrected
`הצג כלל תאונות` to `הצג דיונים ותאונות` as requested

@danielhers Ready to merge

galraij commented:
But the discussion markers are not visible in the list... At any selection... Can we fix that?

danielhers commented:
Yes, please open an issue.

On Thu, Jul 2, 2015 at 10:46 AM Gal Raij <notifications@github.com> wrote:

> But the discussion markers are not visible in the list... At any
> selection... Can we fix that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hasadna/anyway/pull/314#issuecomment-117947057>.
>

omerxx commented:
@galraij they weren't visible before either. this enhancement is only for filtering them.
Having them listed is another thing
